### PR TITLE
VSCode extension: store unsubmitted cell state in vscode

### PR DESCRIPTION
**Pull request by pankgeorg**
Fixes https://github.com/JuliaComputing/pluto-vscode/issues/6. Also rebases to main.

### CODE REVIEW

**Comment by fonsp** [comment]
Cool! Can you rebase in a separate commit? (You have write access) I can't read this PR

**Comment by pankgeorg** [comment]
> Cool! Can you rebase in a separate commit? (You have write access) I can't read this PR

Yeap, sorry!! Can you check again? Also fixed some quoting stuff.

File: frontend/common/VSCodeApi.js [hunk 1]
```diff
@@ -2,9 +2,29 @@
...
 export const available = window.acquireVsCodeApi != null
-export const api = available
-    ? // @ts-ignore
-      window.acquireVsCodeApi()
-    : {
-          postMessage: console.error,
-      }
+// @ts-ignore
+const vscode = window.acquireVsCodeApi?.() ?? {}
+const set_local_cell = (cell_id, new_val) => {
```
Comment (fonsp) [inline frontend/common/VSCodeApi.js:7]: Rename this and the next function to something like `store_cell_input_in_vscode_state` and `load_cell_inputs_from_vscode_state`?

**Comment by fonsp** [comment]
Since the VS Code extension supports multiple notebooks, and multiple clients per notebook, this storage should use a unique client ID provided by the extension host (or at least the notebook ID)

**Comment by pankgeorg** [comment]
You're right, I'll fix that. It _should_ be ok for the short-term, as cells are identifiers are "unique". Fix on the way!

**Comment by fonsp** [comment]
Oh hehe that's true

**Pull request merged**